### PR TITLE
DEPR: Remove ABC(Int|Uint|Float)64Index

### PR DESCRIPTION
**Pull request by topper-123**
Progress towards #42717.

### CODE REVIEW

File: pandas/tests/arithmetic/conftest.py [hunk 1]
```diff
@@ -90,9 +86,10 @@ def zero(request):
 
 @pytest.fixture(
     params=[
-        Float64Index(np.arange(5, dtype="float64")),
-        Int64Index(np.arange(5, dtype="int64")),
-        UInt64Index(np.arange(5, dtype="uint64")),
+        # TODO: add more  dtypes here
```
Comment (topper-123) [inline pandas/tests/arithmetic/conftest.py:89]: I've got some problems with arithmetics with non-64-bit indexes. I'd like to take them in a follow-up.

**Comment by topper-123** [comment]
All green now.

File: pandas/tests/arithmetic/test_numeric.py [hunk 2]
```diff
@@ -669,15 +669,15 @@ def test_mul_int_array(self, numeric_idx):
         result = idx * np.array(5, dtype="int64")
         tm.assert_index_equal(result, idx * 5)
 
-        arr_dtype = "uint64" if isinstance(idx, UInt64Index) else "int64"
+        arr_dtype = "uint64" if idx.dtype == np.uint64 else "int64"
```
Comment (phofl) [inline pandas/tests/arithmetic/test_numeric.py:672]: Can we use is_unsigned_integer_dtype here? Ok as follow up

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @topper-123

**Pull request merged**